### ISSUE TITLE
Automatically re-initializes submodules on Pull

### ISSUE THREAD

**Issue by jasonnicholson** [ForkIssues/TrackerWin#2060]
Fork automatically re-initializes submodules that have been de-initialized during a pull command, as shown in the screenshot below. Calling `init` in the `git submodule update` during a Fork Pull command seems like a bug; this seems like it should be `git submodule update --recursive`. A Fork fetch does not use the `--init` on the `git submodule update` command.

![image](https://github.com/fork-dev/TrackerWin/assets/1058191/73243844-d1c8-4e84-bb55-62de7a10b11d)

**Comment by DanPristupov** [comment]
>  Calling `init` in the `git submodule update` during a Fork Pull command seems like a bug;

No, it's intentional to initialize a new submodule which may have been added in the pulled commits.

> A Fork fetch does not use the `--init` on the `git submodule update` command.

Not sure how fetch is related. Fetch doesn't checkout and submodule update is not needed.

P.S. You can disable submodule update in preferences: File -> Properties -> 'Automatically update submodules on checkout'.

**Comment by jasonnicholson** [comment]
Thank you for the quick reply! :)

We want to deinit some submodules. With the current workflow in Fork, there isn't a way to keep them deinitialized.

Commits do not initialize new submodules. The new submodules are added to the .gitmodules files. This does not automatically mean they should be initialized.

**Comment by DanPristupov** [comment]
> We want to deinit some submodules. With the current workflow in Fork, there isn't a way to keep them deinitialized.

Disable submodule update in preferences: File -> Properties -> 'Automatically update submodules on checkout' and manage submodules manually.

**Comment by jasonnicholson** [comment]
> Disable submodule update in preferences: File -> Properties -> 'Automatically update submodules on checkout' and manage submodules manually.

I am aware of this setting. What do you think of adding a setting that removes `--init` from `git submodule update`? My users know Fork. My users don't want to use the command line.
